autorest/codegen/serializers/model_generic_serializer.py

```diff
@@ -30,7 +30,11 @@ def init_args(model: ObjectSchema) -> List[str]:
                 init_args.append(f"self.{prop.name} = None")
             elif prop.is_discriminator:
                 discriminator_value = f"'{model.discriminator_value}'" if model.discriminator_value else None
-                init_args.append(f"self.{prop.name} = {discriminator_value}")
+                if not discriminator_value:
```

Comment:
discriminator is required, the base class will have None (all the time), and it will be a constant (all the time) in sub-classes. Optional[str] sounds incorrect

Reply:
We added optional string because in the case of the main parent, the discriminator is None, so that is the only case where the discriminator is an Optional[str]. [Here's](https://github.com/Azure/autorest.python/blob/autorestv3/test/vanilla/Expected/AcceptanceTests/BodyComplex/bodycomplex/models/_models_py3.py#L221) an example. I'll add a comment to make it more clear. Thanks!

Reply:
Sounds good. We originally had the Optional[str] for the base class, let me switch that to None. Thanks!

Reply:
Just tried to switch it to None for base class definitions of None, but that's throwing a mypy error, because the typing for the base class needs to encompass the typing of the discriminator in both the base class and the children classes. I think it needs to be left as Optional[str] for the base class. Here's the mypy error:

Incompatible types in assignment (expression has type "str", base class "Fish" defined the type as "None")

Reply:
I see, that's ok then